app/CMakeLists.txt

```diff
@@ -15,21 +15,42 @@ macro(SUBDIRLIST result curdir)
   set(${result} ${dirlist})
 endmacro()
 
+option(INSTALL_ALL_ROBOTS "Enable installation of all robots" OFF)
+set(ROBOT_NAME "$ENV{YARP_ROBOT_NAME}" CACHE PATH "Name of your robot")
```

Comment:
Are you sure about this? `ROBOT_NAME` being a [CMake's cache variable](https://cgold.readthedocs.io/en/latest/tutorials/variables/cache.html), the value is set only when the project is first configure, and it is  not affected by following changes to the `YARP_ROBOT_NAME` env variable, that could be quite confusing for users, see for example: https://github.com/robotology/robotology-superbuild/issues/109 .

Reply:
If we are going to do this, I suggest to set `INSTALL_ALL_ROBOTS` to `ON` in the robotology-superbuild.

Reply:
what do you think to remove the `CACHE` keyword, i.e. `set(ROBOT_NAME "$ENV{YARP_ROBOT_NAME}")`? 
In that case, the variable `YARP_ROBOT_NAME` will be read every time the user run `cmake .`

Reply:
I still think is not ideal, just to give you an idea of the reason:
* I fear that users will change `YARP_ROBOT_NAME` and just run the walking controllers without running `cmake`, and wonder why the module is not able to find the configuration file
* What if a user do not have `YARP_ROBOT_NAME` when it is configuring the project, but sets it just when running the program? 

Just to understand, what is the reason behind this? The files take a lot to install because there are a lot of them? Or the problem is that the `cmake` command takes a lot of time given that `yarp_install` also copies the files in the build during the configuration phase?

Reply:
I guess this was to avoid polluting the ``iCubContrib`` folder, but I agree with @traversaro that it may be safer to install them all anyways.

Reply:
@traversaro 
> Just to understand, what is the reason behind this? The files take a lot to install because there are a lot of them? Or the problem is that the cmake command takes a lot of time given that yarp_install also copies the files in the build during the configuration phase?

As @S-Dafarra wrote, the idea was to avoid copying files that are not necessary.

Reply:
As @traversaro suggested, `INSTALL_ALL_ROBOTS` is `ON` as default